contracts/testing/test_request.go

```diff
@@ -16,6 +16,8 @@ type TestRequest interface {
 	WithToken(token string, ttype ...string) TestRequest
 	WithBasicAuth(username, password string) TestRequest
 	WithoutToken() TestRequest
+	WithSession(attributes map[string]any) TestRequest
+	WithSessionId(sessionId string, attributes map[string]any) TestRequest
```

Comment:
Using session facade

Reply:
Just thinking if the `WithSessionId` is necessary, actually, I don't know when and how to use this method.

Reply:
Let's say you have multiple request to make in one session so If you use `WithSession` only you will lost that session after first request. But if need to use that session again then you need to know sessionId. So create a session and you will get a sessionId and that will be used in multiple requests. But for simplicity this can directly accept an instance of session interface.

Reply:
Thanks for your explanation, could you write an example for your description? Let's check if the code is expected.